icu4c/source/i18n/unicode/numfmt.h

```diff
@@ -266,15 +267,15 @@ class U_I18N_API NumberFormat : public Format {
      * @return clone, or nullptr if an error occurred
      * @stable ICU 2.0
      */
-    virtual NumberFormat* clone() const = 0;
+    virtual NumberFormat* clone() const override { UPRV_UNREACHABLE; return nullptr; }
```

Comment:
This push the compile/build time error checking of the subclass of NumberFormat not implementing a clone method to a runtime assertion / error check and I am concern about this.

Reply:
virtual NumberFormat* clone() const override = 0;